Lesson16/ecenaz/script.js

```diff
@@ -0,0 +1,490 @@
+averageArray([5,10,15,20,25,30]);
+/*
+22. Flatten a 2D Array (Nested Loops)
+    - Define a function `flattenArray(twoDArray)` that takes an array of arrays
+      (e.g., [[1,2],[3,4]]) and uses nested loops to create a new one-dimensional array.
+    - Log the flattened array.
+*/
+function flattenArray(twoDArray){
+  let flattenedArray= [];
+  for (let i=0; i<twoDArray.length; i++){
+    for (let j=0; j<twoDArray.length; j++){
+      flattenedArray.push(twoDArray[i][j]);
+    }
+  }
+  console.log(flattenedArray);
+}
+
+flattenArray([[1,2],[3,4]]);
+/*
+23. Find Words Containing a Letter
+    - Define a function `findWordsWithLetter(words, letter)` that loops through
+      an array of words and returns a new array of only the words that contain
+      the given letter.
+    - Log the filtered array.
+*/
+function findWordsWithLetter(words, letter){
+  let filteredArray=[];
+for (let i=0; i<words.length; i++){
+  if(words[i].includes(letter)){
+    filteredArray.push(words[i]);
+  }
+}
+ console.log(filteredArray);
+}
+findWordsWithLetter(["elma", "armut","ananas","avokado","muz","turp"],"a");
+
+/*
+24. Push and Pop Operations
+    - Define a function `pushPopExample(arr, itemToPush)` that:
+      - pushes itemToPush to arr
+      - logs the updated array
+      - then pops the last element
+      - logs the popped element
+      - logs the final array
+*/
+function pushPopExample(arr, itemToPush){
+
+  for(let i=0; i<itemToPush.length; i++){
+    arr.push(itemToPush[i]);
+    console.log("Updated arrays",arr); 
+  }
+  for (let i=0; i<itemToPush; i++){
+    let poppedItem = arr.pop();
+    console.log("Popped element:",arr);
+    console.log("Final array:", arr);
+  }
+}
+
+pushPopExample([1,2,3,4], [5]);
+/*
+25. Push and Shift Operations
+    - Define a function `manageQueue(queue, newPerson)` that:
+      - push `newPerson` to the end of `queue`
+      - logs the updated queue
+      - shifts (removes) the first person in the queue
+      - logs the removed person
+      - logs the final queue
+*/
+function manageQueue(queue, newPerson){
+  for(let i=0; i<newPerson.length; i++){
+   queue.push(newPerson[i]); 
+   console.log("Updated queue:", newPerson);
+  }
+  for(let i=0; i<newPerson.length; i++){
+    let removedPerson = queue.shift();
+    console.log("Removed person:" ,removedPerson);
+console.log("Finale queue:", queue);
+  }
+  
+}
+manageQueue(["Ece","Naz","Zagli"], ["Benim","Adım"]);
+/*
+26. To-Do List Application 
+  - Define a function `updateTodoList(todoList, startIndex, deleteCount, ...newTasks)`:
+   - Logs the current list of tasks.
+   - Removes `deleteCount` tasks starting at `startIndex`.
+   - Inserts any new tasks at the end of the array.
+   - Logs the updated list.
+*/
+
+const todoList = ["Study JS", "Eat breakfast", "Walk dog"];
+function updateTodoList(todoList, startIndex, deleteCount, ...newTasks){
+  console.log("Current list:", todoList);
+
+  let index= startIndex;
+  for (let i=0; i<deleteCount; i++  ){
```

Comment:
The `splice(startIndex, deleteCount)` method can remove multiple items from an array in a single step. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/splice

You don’t need to loop through each item and remove it one by one because splice already handles this for you.